src/audio/audio_context.rs

```diff
@@ -0,0 +1,37 @@
+//! Provides a context for loading a file.
+
+use super::source::Source;
+use assets::*;
+
+use std::sync::Arc;
+
+/// A context for loading audio files
+pub struct AudioContext {
+    inner: SimpleContext<Source, Vec<u8>, NoError, fn(Vec<u8>) -> Result<Source, NoError>>,
```

Comment:
@torkleyy Just to clarify: Dynamic dispatch usually refers to a vtable lookup occurring, this doesn't involve a vtable lookup though, it just uses a hard coded function pointer so this isn't really that expensive.